entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityOperationService.kt

```diff
@@ -103,27 +124,42 @@ class EntityOperationService(
         }
 
         return try {
-            val (_, notUpdated) = entityService.appendEntityAttributes(
-                entity.id,
-                entity.attributes,
-                false
-            )
-
-            if (notUpdated.isEmpty()) {
-                Either.right(entity.id)
-            } else {
-                Either.left(
-                    BatchEntityError(
-                        entity.id,
-                        ArrayList(notUpdated.map { it.attributeName + " : " + it.reason })
-                    )
-                )
-            }
+            processor(entity)
         } catch (e: BadRequestDataException) {
             Either.left(BatchEntityError(entity.id, arrayListOf(e.message)))
         }
     }
 
+    /*
+     * Transactional because it should not delete entity attributes if new ones could not be appended.
+     */
+    @Transactional(rollbackFor = [BadRequestDataException::class])
+    @Throws(BadRequestDataException::class)
+    private fun replaceEntity(entity: ExpandedEntity): Either<BatchEntityError, String> {
+        neo4jRepository.deleteEntityAttributes(entity.id)
+        entityService.appendEntityAttributes(entity.id, entity.attributes, false)
```

Comment:
I'm wondering if a check that updated.size == entity.attributes.size would make sense?

Reply:
Indeed it would make sense, and I think it should be done.